pkg/asset/ibi/ibi_config.go

```diff
@@ -0,0 +1,131 @@
+package ibi
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/types/ibi"
+)
+
+var ibiConfigFilename = "ibi-config.yaml"
+
+// ImageBasedInstallConfig reads the ibi-config.yaml file.
+type ImageBasedInstallConfig struct {
+	File     *asset.File
+	Config   *ibi.Config
+	Template string
+}
+
+var _ asset.WritableAsset = (*ImageBasedInstallConfig)(nil)
+
+// Name returns a human friendly name for the asset.
+func (*ImageBasedInstallConfig) Name() string {
+	return "Image-based Install Config"
+}
+
+// Dependencies returns all of the dependencies directly needed to generate
+// the asset.
+func (*ImageBasedInstallConfig) Dependencies() []asset.Asset {
+	return []asset.Asset{}
+}
+
+// Generate generates the Image-based Install Config manifest.
+func (i *ImageBasedInstallConfig) Generate(dependencies asset.Parents) error {
+	ibiConfigTemplate := `#
+# Note: This is a sample ImageBasedInstallConfig file showing
+# which fields are available to aid you in creating your
+# own ibi-config.yaml file.
+#
+apiVersion: v1beta1
+kind: ImageBasedInstallConfig
+metadata:
+  name: example-ibi-config
+  namespace: cluster0
+seedImage: quay.io/openshift-kni/seed-image:4.16.0
+seedVersion: 4.16.0
+pullSecretFile: ./pull-secret.json
+authFile: ./auth-file.json
+sshPublicKeyFile: ./id_rsa.pub
+rhcosLiveIso: https://mirror.openshift.com/pub/openshift-v4/amd64/dependencies/rhcos/latest/rhcos-live.x86_64.iso
+installationDisk: vda
```

Comment:
This should definitely be a path to the device (`/dev/vda`) because the right thing to do in almost all cases will be to pass a `/dev/disk/by-path/...` alias.